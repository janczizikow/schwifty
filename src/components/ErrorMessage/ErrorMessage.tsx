import React from 'react';
import {Text, View, Button, StyleSheet} from 'react-native';

interface OwnProps {
  onRetry?: () => void;
}

const ErrorMessage: React.FC<OwnProps> = ({onRetry}) => (
  <View style={styles.root}>
    <View style={styles.centerText}>
      <Text>Download Error</Text>
      <Text>Check your internet settings and retry.</Text>
    </View>
    {onRetry && <Button title="Retry" onPress={onRetry} />}
  </View>
);

const styles = StyleSheet.create({
  root: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    marginHorizontal: 10,
  },
  centerText: {
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: 16,
  },
});

export default ErrorMessage;
