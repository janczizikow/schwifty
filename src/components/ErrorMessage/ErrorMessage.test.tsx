import {Button} from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ErrorMessage from './ErrorMessage';

describe('<ErrorMessage />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ErrorMessage />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a Button when given a prop', () => {
    const testRenderer = renderer.create(
      <ErrorMessage onRetry={() => jest.fn()} />,
    );
    expect(testRenderer.root.findAllByType(Button)).toHaveLength(1);
  });
});
