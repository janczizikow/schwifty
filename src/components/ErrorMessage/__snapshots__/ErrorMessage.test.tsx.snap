// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginHorizontal": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 16,
      }
    }
  >
    <Text>
      Download Error
    </Text>
    <Text>
      Check your internet settings and retry.
    </Text>
  </View>
</View>
`;
